include/swift/AST/DiagnosticsSema.def

```diff
@@ -2710,6 +2710,10 @@ NOTE(missing_member_type_conformance_prevents_synthesis, none,
      "protocol %2, preventing synthesized conformance "
      "of %3 to %2",
      (unsigned, Type, Type, Type))
+NOTE(classes_automatic_protocol_synthesis,none,
+    "automatic synthesis of '%0' is not supported for classes "
+    "because of inheritance.",
```

Comment:
I don’t think we need to mention this (it’s not clear or explained how inheritance affects the synthesis), so I think you should just drop the “because of inheritance” bit. Also, please remove the period from the end of the diagnostic text as well.

Reply:
Oh, I didn’t read the bug report properly. We need to provide a reason! Please ignore what I said earlier about dropping the inheritance bit. However, I think we either need to drop it for `final` classes (since you can’t inherit it).

Reply:
Lets keep it simple and just drop an inheritance part of diagnostics message.

Reply:
Fixed.